benchmarking/smartseq2/confounding_factor_analysis/Confounding_factor_analysis.R

```diff
@@ -0,0 +1,325 @@
+  p.sub <- pmat[(nc1 + 1):nc2, 1:nc1]
+  # rank column by # of significant correlation test between qc metrics and PCs
+  rank.p <- order(colSums(p.sub < 0.05, na.rm = T), decreasing = T)
+  png(
+    paste(output_name, '_corrplot_top_', npcs, '_vs_qc_mets.png', sep = ''),
+    width = 5000,
+    height = 5000
+  )
+  corrplot(
+    t(M.sub[, rank.p]),
+    p.mat = t(p.sub[, rank.p]),
+    is.corr = TRUE,
+    insig = "label_sig",
+    sig.level = c(.001, .01, .05),
+    pch.cex = 1.5,
+    pch.col = "white",
+    na.label = " ",
+    tl.cex = 5,
+    cl.cex = 5,
+    tl.col = "black",
+    method = "color",
+    col = colorRampPalette(c("blue", "white", "red"))(200)
+  )
+  dev.off()
+  # use the full correlation matrix
+  # rank column/row by # of significant correlation test between qc metrics vs PCs
+  rank.p <- order(rowSums(pmat < 0.05, na.rm = T), decreasing = T)
+  png(
+    paste(
+      output_name,
+      '_corrplot_top_',
+      npcs,
+      '_vs_qc_mets_all.png',
+      sep = ''
+    ),
+    width = 8000,
+    height = 8000
+  )
+  corrplot(
+    t(M[rank.p, rank.p]),
+    p.mat = t(pmat[rank.p, rank.p]),
+    is.corr = TRUE,
+    insig = "label_sig",
+    sig.level = c(.001, .01, .05),
+    pch.cex = 1.5,
+    pch.col = "white",
+    na.label = " ",
+    tl.cex = 5,
+    cl.cex = 5,
+    tl.cor = 'black',
+    method = "color",
+    col = colorRampPalette(c("blue", "white", "red"))(200)
+  )
+  dev.off()
+  # only return the subset of correlation test results, which is qc metrics vs PCs only
+  return(p.sub)
+}
+# add extra theme
+addTheme <- function(p) {
+  p <- p + theme(legend.position = "top")
+  p <-
+    p + theme(plot.title = element_text(
+      hjust = 0.5,
+      size = 20,
+      face = 'bold'
+    ))
+  p <-
+    p + theme(
+      axis.title.x = element_text(
+        color = 'black',
+        size = 18,
+        face = 'bold'
+      ),
+      axis.title.y = element_text(
+        size = 18,
+        color = 'black',
+        face = 'bold'
+      )
+    )
+  p <-
+    p + theme(
+      axis.text.y = element_text(
+        color = 'black',
+        size = 18,
+        face = 'bold'
+      ),
+      axis.text.x = element_text(
+        color = 'black',
+        size = 18,
+        face = 'bold'
+      )
+    )
+  p <-
+    p + theme(legend.text = element_text(size = 15, face = 'bold'),
+              legend.title = element_blank())
+  return(p)
+}
+# xy-plot: # significant correlation between QC and data
+# x-axis is the # significant test from one pipeline
+# y-axis is the # significant test form another pipeline
+# inputs are correlation tesxt matrix, such as output of CorrQCvsPCs()
+plotSignCorr <- function(base_mat, updated_mat, output_name) {
+  # sum significant test
+  sum.sig1 <- colSums(base_mat < 0.05, na.rm = T)
+  sum.sig2 <- colSums(updated_mat < 0.05, na.rm = T)
+  # merge by qc metrics names
+  df <- merge(sum.sig1, sum.sig2, by = 0)
+  colnames(df) <- c('QC', 'Base', 'Updated')
```

Comment:
Why do you need to rename the columns here?

Reply:
it is not rename. it is just name of two columns, otherwise the two columns names are `sum.sig1` and `sum.sig2` and does not really mean anything

Reply:
it is to give the column names, otherwise the column names are `sum.sig1` and `sum.sig2`